rmgpy/rmgobject.pyx

```diff
@@ -54,7 +54,9 @@ cdef class RMGObject(object):
         all_attributes = [attr for attr in dir(self) if not attr.startswith('_')]
         for attr in all_attributes:
             val = getattr(self, attr)
-            if val is not None and not callable(val) and val != '':
+            if isinstance(val, np.ndarray): # Check this first so that we don't try to do elementwise comparisons
+                output_dict[attr] = expand_to_dict(val)
+            elif val is not None and not callable(val) and val != '':
```

Comment:
If you're doing the same thing in both cases, you can just put the ndarray check first with an or.